Doctrine/Phpcr/SitemapUrlInformationProvider.php

```diff
@@ -0,0 +1,163 @@
+     * @param DocumentManager $manager
+     * @param RouterInterface $router
+     * @param SecurityContextInterface $publishWorkflowChecker
+     * @param string $defaultChanFrequency
+     * @param PsrLogger $logger
+     * @param SeoPresentationInterface $seoPresentation
+     */
+    public function __construct(
+        DocumentManager $manager,
+        RouterInterface $router,
+        $defaultChanFrequency,
+        PsrLogger $logger,
+        SeoPresentationInterface $seoPresentation,
+        SecurityContextInterface $publishWorkflowChecker = null
+    ) {
+        $this->manager = $manager ;
+        $this->router = $router;
+        $this->defaultChanFrequency = $defaultChanFrequency;
+        $this->logger = $logger;
+        $this->seoPresentation = $seoPresentation;
+        $this->publishWorkflowChecker = $publishWorkflowChecker;
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    public function getUrlInformation()
+    {
+        $routeInformation = array();
+
+        // todo rewrite query when https://github.com/symfony-cmf/CoreBundle/issues/126 is ready
+        $contentDocuments = $this->manager->createQuery(
+            "SELECT * FROM [nt:unstructured] WHERE (visible_for_sitemap = true)",
+            QueryInterface::JCR_SQL2
+        )->execute();
+
+        foreach ($contentDocuments as $document) {
+            if (null != $this->publishWorkflowChecker &&
+                !$this->publishWorkflowChecker->isGranted(array(PublishWorkflowChecker::VIEW_ATTRIBUTE), $document)
+            ) {
+                continue;
+            }
+
+            try {
+                $routeInformation[] = $this->computeUrlInformationFromContent($document);
+            } catch (\Exception $e) {
+                $this->logger->info($e->getMessage());
+            }
+        }
+
+        return $routeInformation;
+    }
+
+    /**
+     * To generate the url information object out of the content this method
+     * extracts the data from the given content.
+     *
+     * @param $content
+     *
+     * @return UrlInformation
+     */
+    protected function computeUrlInformationFromContent($content)
+    {
+        $urlInformation = new UrlInformation();
+        $urlInformation->setLoc($this->router->generate($content, array(), true));
+        $urlInformation->setChangeFreq($this->defaultChanFrequency);
+
+        if ($this->alternateLocaleProvider) {
+            $collection = $this->alternateLocaleProvider->createForContent($content);
+            $urlInformation->setAlternateLocales($collection->toArray());
+        }
+
+        $seoMetadata = $this->seoPresentation->getSeoMetadata($content);
+        if (null !== $seoMetadata->getTitle()) {
+            $urlInformation->setLabel($seoMetadata->getTitle());
+        }
+
+        return $urlInformation;
+    }
+
+    /**
+     * @param AlternateLocaleProviderInterface $alternateLocaleProvider
+     */
+    public function setAlternateLocaleProvider($alternateLocaleProvider)
```

Comment:
but you could just alter the service definition to add one more constructor argument then, instead of adding a method call. but then its not super important either, i am ok with keeping as is.